src/sentry/api/endpoints/organization_events_trace.py

```diff
@@ -1325,9 +1358,11 @@ def serialize(
                 warning_extra,
                 event_id,
                 detailed,
-                allow_orphan_errors,
             )
             return results
+
+        # Code past here is deprecated, but must continue to exist until sentry installs in every possible environment
+        # are able to support storing span data
```

Comment:
We need to be more strict here. As long as any environment has the option to turn off spans, we have to keep this code around.

Reply:
Hmm what do you mean by being more strict?

Reply:
I guess what I'm trying to say is that it is not enough that every environment is capable of storing span data. Every environment must only use span data before this old code can be removed.

Reply:
Reworded this slightly